### ISSUE TITLE
Custom Theme Variables

### ISSUE THREAD

**Issue by MinusGix** [lapce/lapce#601]
The #600 PR changed how themes work, mostly for the better. However, it does lose the ability to declare a variable, ex `background` and then use that as the value for various parts of the UI. I think it would be likely useful to reintroduce this, and show it in the UI so that theme-authors can have variables that let the user customize a good portion of the UI (and if the user wants something other than the variable, they just change it).

**Comment by dzhou121** [comment]
Good point. It's kind of hard coded for the "base" color names right now. We could use a HashMap for it.

**Comment by derspyy** [comment]
this is incredibly valuable for implementing theme palettes, such as [rosé pine](https://rosepinetheme.com), [catppuccin](https://catppuccin.com) and [nord](https://www.nordtheme.com) _and_ maintain their continuing support since updating them will be easier.
